src/sentry/api/endpoints/internal_register_trusted_relay.py

```diff
@@ -0,0 +1,105 @@
+from datetime import datetime, timezone
+
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.authentication import UserAuthTokenAuthentication
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.api.bases.organization import OrganizationPermission
+from sentry.api.serializers.models.organization import TrustedRelaySerializer
+from sentry.models.options.organization_option import OrganizationOption
+from sentry.models.organization import Organization
+
+
+class TrustedRelayPermission(OrganizationPermission):
+    scope_map = {
+        "GET": ["org:read", "org:write", "org:admin"],
+        "POST": ["org:write", "org:admin"],
+        "PUT": ["org:write", "org:admin"],
+        "DELETE": ["org:admin"],
+    }
+
+
+@control_silo_endpoint
+class InternalRegisterTrustedRelayEndpoint(Endpoint):
+    publish_status = {
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.OWNERS_INGEST
+    authentication_classes = (UserAuthTokenAuthentication,)
+    permission_classes = (TrustedRelayPermission,)
+
+    def post(self, request: Request) -> Response:
+        """
+        Register a new trusted relay for an organization.
+        If a relay with the given public key already exists, update it.
+        """
+        organization_id = request.auth.organization_id
+        if not organization_id:
+            return Response(
+                {"detail": "Organization not found in the request"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            organization = Organization.objects.get(id=organization_id)
```

Comment:
You can't use `Organization` from control silo. You'd need to fetch the organization from the `OrganizationService`. Given that all your reads and writes are to region resources, you will have a simpler time if this were a region endpoint.